modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusion.C

```diff
@@ -52,15 +52,24 @@ PINSFVMomentumDiffusion::computeStrongResidual()
 
   const auto elem_face = elemFromFace();
   const auto neighbor_face = neighborFromFace();
-  const auto mu_elem = _mu(elem_face);
-  const auto mu_neighbor = _mu(neighbor_face);
-  const auto eps_elem = has_elem ? _eps(elem_face) : _eps(neighbor_face);
-  const auto eps_neighbor = has_neighbor ? _eps(neighbor_face) : _eps(elem_face);
 
   // Compute the diffusion driven by the velocity gradient
   // Interpolate viscosity divided by porosity on the face
   ADReal mu_face;
-  interpolate(Moose::FV::InterpMethod::Average, mu_face, mu_elem, mu_neighbor, *_face_info, true);
+
+  const auto mu_elem = has_elem ? _mu(elem_face) : _mu(neighbor_face);
+  const auto eps_elem = has_elem ? _eps(elem_face) : _eps(neighbor_face);
+
+  ADReal mu_neighbor;
+  ADReal eps_neighbor;
```

Comment:
remove and put it inside if?

Reply:
We can't  do it. This part at the end of the function uses it too so it cannot be defined in an if-else statement:
```
ADReal coeff_face;
  const auto coeff_elem = mu_elem / eps_elem * _var(elem_face);
  if (onBoundary(*_face_info))
    coeff_face = coeff_elem;
  else
  {
    const auto coeff_neighbor = mu_neighbor / eps_neighbor * _var(neighbor_face);
    interpolate(Moose::FV::InterpMethod::Average,
                coeff_face,
                coeff_elem,
                coeff_neighbor,
                *_face_info,
                true);
  }
```